chia/full_node/mempool_check_conditions.py

```diff
@@ -156,6 +158,45 @@ def get_spends_for_block(generator: BlockGenerator) -> List[CoinSpend]:
 
     return spends
 
+def get_spends_for_block_with_conditions(generator: BlockGenerator, height: int, constants: ConsensusConstants) -> List[Dict[str, Any]]:
+    args = bytearray(b"\xff")
+    args += bytes(DESERIALIZE_MOD)
+    args += b"\xff"
+    args += bytes(Program.to([bytes(a) for a in generator.generator_refs]))
+    args += b"\x80\x80"
+
+    flags = get_flags_for_height_and_constants(height, constants)
+
+    _, ret = run_chia_program(
+        bytes(generator.program),
+        bytes(args),
+        DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM,
+        flags,
+    )
+
+    spends = []
+
+    for spend in Program.to(ret).first().as_iter():
+        parent, puzzle, amount, solution = spend.as_iter()
+        puzzle_hash = puzzle.get_tree_hash()
+        _, r = run_chia_program(
+            puzzle.as_bin(),
+            solution.as_bin(),
+            DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM,
+            flags,
+        )
+        spends.append({
+            "coin": {
+                "amount": int_from_bytes(amount.atom),
+                "parent_coin_info": parent.atom,
+                "puzzle_hash": puzzle_hash,
+            },
+            "puzzle": puzzle,
+            "solution": solution,
+            "conditions": Program.to(r),
```

Comment:
this call is potentially *very* expensive. Do we always trust callers of this function?
would it not be acceptable to return a condensed version of the conditions here? e.g. Something like what's returned by `run_block_generator2()`.

if you call this function on a malicious block, your node will run out of memory.

Reply:
The program is run with `MAX_BLOCK_COST_CLVM` and the code is coming from a full block in the block store, so hasn't it already been run at least once before anyway?

Reply:
If there are duplicate conditions we may want to see/display that.